pages/api/personalization/get-user-data.js

```diff
@@ -0,0 +1,9 @@
+export default function handler(req, res) {
+  // This API route accepts only POST requests.
+  if (req.method !== 'POST') {
```

Comment:
Why not `ensurePostRequest` here?

Reply:
I've added this file by mistake, it will be used for the next feature (remembering dark mode). Removed it for now.